docker/centos/Dockerfile

```diff
@@ -0,0 +1,43 @@
+# This stage builds an rpm from the source
+FROM centos:centos7 as centos-builder
```

Comment:
why do we need a build here? we have RPMs already https://github.com/FRRouting/frr/releases/download/frr-7.2/frr-7.2-01.el7.centos.x86_64.rpm

Reply:
Thank you for comment.
I think that it is better to be able to use Docker not only as an environment for running FRR but also as an environment for building RPM easily. For example, if we apply a patch and want to test it in a real environment, I think it can be used as a way to build an RPM.
Also, FRR doesn't have a yum repository like https://deb.frrouting.org/, so if we install RPM from https://github.com/FRRouting/frr/releases we need to change Dockerfile every release.

Reply:
That's fine, but we already use libyang and librtr with sticky versions, that's almost the same as installing from the RPMs we already have. This is just my opinion.

Reply:
IMO better to build from source where possible. I would prefer the ability to build a "latest" docker image off master instead of gating docker images off our release cycle.

Reply:
@qlyoung I think the best way would be to spawn RPM repository the same way as we have with Debian/Ubuntu (no for this, but in the future). 

With Debian docker we use our compiled version, with Alpine from the source. Somehow not consistent :) 

I'm fine with this change :)

Reply:
Actually the Debian docker images were taken as a quick and dirty patch with the intent to convert them to a source build in the immediate future. Unfortunately the person who submitted the change got busy with other things and has never gotten around to it. The Debian docker image needs to be changed to a source build as well.

The point of these tree-integrated docker images is for development work and testing (out) FRR, not production deployment. Having dockerfiles that produce images with binary FRR packages is effectively providing a production image.

So I still prefer a source build.

Reply:
:+1: